### DISCUSSION TITLE
how to encrypt existing objects in minio bucket using mc mv command with a SSE-KMS key

### DISCUSSION THREAD

**Discussion by Archvec** [minio/minio#18851] [Q&A] [answered]
Hi,

I have a working minio cluster on premise that use KES server with keys on filesystem (all minio/mc/kes version are up to date from january 2024)
To setup automatic encryption of my bucket, i have setup SSE-KMS according to the following documentation : https://min.io/docs/minio/linux/reference/minio-mc/mc-encrypt-set.html

it works properly however at the end of that documentation it specify the existing objects in bucket won't be encrypted. To achieve this use mc mv --encrypt" command.
I don't understand the appropriate syntax to specify my SSE-KMS encryption key in the "mc mv --encrypt" commands. It is possible to use SSE-KMS? if yes do you have an example of syntax?

```
[minio@test ~]$ mc stat myminio/test --insecure
Name      : test
Date      : 2024-01-22 17:31:44 CET
Size      : N/A
Type      : folder

Properties:
  Encryption:
        Key Type: SSE-KMS
        Key ID: my-key-1
  Versioning: Un-versioned
  Location: us-east-1
  Anonymous: Disabled
  ILM: Disabled
```

Thanks

**Comment by harshavardhana** [comment, marked as answer]
That is not allowed @Archvec - retroactive encryption is not possible. Create a new bucket, enable encryption, and then do `mc mirror` bucket to bucket, and then delete the previous bucket once complete.

**Reply by Archvec** [reply]
Hi, yes retroactive is not possible but this end of this page : https://min.io/docs/minio/linux/reference/minio-mc/mc-encrypt-set.html
state that it must be done manually. To achieve it manually only SSE-S3 method is support (not SS-KMS) , isn't it ?

**Reply by Archvec** [reply]
this page is unclear because it explain how to setup encryption at the bucket level using sse-kms then state about the behavior for existing data. if it's not possible using mc mv, the "behavior" should be to mirror the bucket as you've proposed
